internal/tools/cloudflare/dns.go

```diff
@@ -114,42 +114,64 @@ func (c *Client) CreateDNSRecords(dnsNames []string, dnsEndpoints []string, logg
 
 		ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
 		defer cancel()
-		recordResp, err := c.cfClient.CreateDNSRecord(ctx, zoneID, record)
+
+		existingRecords, err := c.cfClient.DNSRecords(ctx, zoneID, cf.DNSRecord{
+			Name: dnsName,
+			Type: "CNAME",
+		})
 		if err != nil {
-			// We do not have a clear indication if the record already exists
-			// based only on the status code (400 may mean invalid input),
-			// therefore we also need to compare the error code with the UNDOCUMMENTED
-			// error code that refers to a record already existing.
-			cloudflareRequestError := &cf.RequestError{}
-			if !errors.As(err, &cloudflareRequestError) || !cloudflareRequestError.InternalErrorCodeIs(recordExistsErrorCode) {
+			return errors.Wrap(err, "failed to get existing DNS record at Cloudflare")
+		}
+
+		if len(existingRecords) == 0 {
+			ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+			defer cancel()
+			recordResp, err := c.cfClient.CreateDNSRecord(ctx, zoneID, record)
+			if err != nil {
 				return errors.Wrap(err, "failed to create DNS Record at Cloudflare")
 			}
-			log.Info("Record already exists, trying to update...")
-			err = c.updateDNSRecord(zoneID, record, log)
+
+			log.Debugf("Cloudflare create DNS record response: %v", recordResp)
+			return nil
+		}
+
+		recordToUpdate := existingRecords[0]
+		doUpdate := false
+
+		if recordToUpdate.Content != record.Content || recordToUpdate.TTL != record.TTL {
+			doUpdate = true
+		}
+		if recordToUpdate.Proxied != nil && record.Proxied != nil {
+			if *recordToUpdate.Proxied != *record.Proxied {
+				doUpdate = true
+			}
+		} else if recordToUpdate.Proxied != nil || record.Proxied != nil {
```

Comment:
Shouldn't this be: ?

```suggestion
		} else if recordToUpdate.Proxied != record.Proxied {
```